pkg/otlp/metrics/origin.go

```diff
@@ -14,11 +14,198 @@
 
 package metrics
 
+import (
+	"fmt"
+	"strings"
+)
+
 // OriginProduct defines the origin product.
 type OriginProduct int32
 
+const (
+	OriginProductUnknown OriginProduct = 0
+	// OriginProductDatadogAgent is the origin for metrics coming from the Datadog Agent OTLP Ingest.
+	OriginProductDatadogAgent OriginProduct = 10
+	// OriginProductDatadogExporter is the origin for metrics coming from the OTel Collector Datadog Exporter.
+	OriginProductDatadogExporter OriginProduct = 17
+)
+
+func (o OriginProduct) String() string {
+	switch o {
+	case OriginProductUnknown:
+		return "unknown"
+	case OriginProductDatadogAgent:
+		return "datadog-agent"
+	case OriginProductDatadogExporter:
+		return "datadog-exporter"
+	default:
+		return fmt.Sprintf("OriginProduct(%d)", o)
+	}
+}
+
 // OriginCategory defines the origin category.
 type OriginCategory int32
 
+// OriginCategoryOTLP is the origin category for all metrics coming from OTLP.
+// All metrics produced by the translator MUST have origin category set to OTLP.
+const OriginCategoryOTLP OriginCategory = 17
+
+func (o OriginCategory) String() string {
+	switch o {
+	case OriginCategoryOTLP:
+		return "otlp"
+	default:
+		return fmt.Sprintf("OriginCategory(%d)", o)
+	}
+}
+
 // OriginService defines the origin service.
 type OriginService int32
+
+// List all receivers that set the scope name.
```

Comment:
I think this should also contain `connectors` ? A few add the scope: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/connector/countconnector/connector.go#L79.

Reply:
Spoke offline about this, makes sense but will add it later since it needs to go through backend approval